text/0000-build-systems.md

```diff
@@ -0,0 +1,446 @@
+extensibility points in Cargo, but rather to outline a general plan for how to
+get there over time. Individual components that add significant features to
+Cargo will need follow-up RFCs before stabilization.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+The plan proposed in this RFC is to address the two use-cases from the
+[motivation] section in parallel:
+
+- **For the mixed build system case**, we will triage feature requests and work on
+  adding further points of extensibility to Cargo based on expected impact. Each
+  added point of extensibility should ease build system integration for another
+  round of customers.
+
+- **For the homoegenous build system case**, we will immediately pursue
+  extensibility points that will enable the external build system to perform
+  many of the tasks that Cargo does today--but while still meeting our
+  interoperability constraints. We will then work on smoothing remaining rough
+  edges, which have a high degree of overlap with the work on mixed build
+  systems.
+
+In the long run, these two parallel lines of work will converge, such that we
+offer a complete spectrum of options (in terms of what Cargo controls versus an
+external system). But they start at critically different points, and working on
+those in parallel is the key to delivering value quickly and incrementally.
+
+## A high-level model of what Cargo does
+
+Before delving into the details of the plan, it's helpful to lay out a mental
+model of the work that Cargo does today, broken into several stages:
+
+| **Step**              | **Conceptual output**                                                                                                                                    | **Related concerns**             |
+| --------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------------- |
+| Dependency resolution | Lock file                                                                                                                                                | Custom registries, mirrors, offline/local, native deps, ... |
+| Build configuration   | Cargo settings per crate in graph                                                                                                                        | Profiles                            |
+| Build lowering        | A build plan: a series of steps that must be run in sequence, including rustc and binary invocations | Build scripts, plugins              |
+| Build execution       | Compiled artifacts                                                                                                                                       | Caching                             |
+
+The first stage, dependency resolution, is the most complex; it's where our
```

Comment:
Any chance you could avoid the use of first-person pronouns here? "We" and "our" are a little confusing.

"it's where Cargo's model of semver comes into play" etc seem clearer.